dbus/methods.go

```diff
@@ -417,6 +417,24 @@ func (c *Conn) listUnitsInternal(f storeFunc) ([]UnitStatus, error) {
 	return status, nil
 }
 
+func (c *Conn) getUnitInternal(f storeFunc) (string, error) {
+	var result dbus.ObjectPath
+
+	err := f(&result)
+
+	// Nothing in this library actually accepts a dbus.ObjectPath, so it's much
+	// more useful as a name
+	name := unitName(result)
+
+	return name, err
+}
+
+// GetUnitByPIDContext returns the unit name for a given PID. The PID must refer
+// to an existing system process
+func (c *Conn) GetUnitByPIDContext(ctx context.Context, pid uint32) (string, error) {
```

Comment:
We can drop the `Context` suffix from here (it's an historical artifact added for method that started their existence before `context` was a thing).

I agree getting the unit name is probably the most common case, but the object path could be useful too in some case.
I'd suggest having a plain `GetUnitByPID` that directly returns the object path, and an additional `GetUnitNameByPID` which internally does some translation. No need for a separate `getUnitInternal` helper, that logic can be folded directly into the latter.

Reply:
Works for me, should be addressed